test/integration/test_genomes.py

```diff
@@ -0,0 +1,69 @@
+from galaxy.exceptions import (
+    MissingDataError,
+    ObjectNotFound,
+)
+from galaxy.visualization.genomes import Genome
+from galaxy_test.driver import integration_util
+
+
+class MockGenome:
```

Comment:
I would feel better if instead of using a mock and breaking into app - you tweaked the actual configuration that Galaxy loads in order to produce this. I'll admit it is a personal preference though. I suspect we don't have any test coverage of `` app.genomes``.

Reply:
No, you're right - it's an objectively better approach. I wasn't sure how difficult it would be to setup the required input data in this case - not so difficult after all. 

There is no test coverage of `app.genomes`, and, in my opinion, in its current form it is mostly untestable. Making it testable requires refactoring, which requires adding unit tests for its core functionality - and that may be a time-consuming task. Assuming this is not high priority, I'd suggest leaving it as is: the current integration tests cover the most general use cases of the available endpoints, including the corrected error handling. Although I do need to add tests for the 2 remaining endpoints which currently result in errors that are not properly handled - usegalaxy.[org|eu] included.  But if proper test coverage of this module *is* high priority, I'll take this on, of course.